### PR TITLE
Fix lighting module under auto offset

### PR DESCRIPTION
**Pull request by Pessimistress**
For https://github.com/uber/deck.gl/issues/2223

#### Change List
- Rewrite target coordinate system with lnglat offset parameters if in `LNGLAT_AUTO_OFFSET` mode

### CODE REVIEW

File: modules/core/src/shaderlib/lighting/lighting.js [hunk 1]
```diff
@@ -58,25 +59,33 @@ function getUniforms(opts = INITIAL_MODULE_OPTIONS) {
...
     diffuseRatio = DEFAULT_DIFFUSE_RATIO,
     specularRatio = DEFAULT_SPECULAR_RATIO
   } = opts.lightSettings;
 
+  let coordinateSystem = opts.coordinateSystem;
+  let coordinateOrigin = opts.coordinateOrigin;
+
+  if (context.project_uCoordinateSystem === PROJECT_COORDINATE_SYSTEM.LNGLAT_AUTO_OFFSET) {
+    coordinateSystem = COORDINATE_SYSTEM.LNGLAT_OFFSETS;
+    coordinateOrigin = context.project_coordinate_origin;
```
Comment (georgios-uber) [inline modules/core/src/shaderlib/lighting/lighting.js]: Who is providing the context?

Comment (Pessimistress) [reply modules/core/src/shaderlib/lighting/lighting.js]: The `context` parameter contains all uniforms generated by a module's dependencies (in this case `project`).

File: modules/core/src/shaderlib/lighting/lighting.js [hunk 2]
```diff
@@ -58,25 +59,33 @@ function getUniforms(opts = INITIAL_MODULE_OPTIONS) {
 
     lightsPosition = DEFAULT_LIGHTS_POSITION,
     lightsStrength = DEFAULT_LIGHTS_STRENGTH,
-    coordinateSystem = COORDINATE_SYSTEM.LNGLAT,
-    coordinateOrigin = DEFAULT_COORDINATE_ORIGIN,
+    coordinateSystem: fromCoordinateSystem = COORDINATE_SYSTEM.LNGLAT,
+    coordinateOrigin: fromCoordinateOrigin = DEFAULT_COORDINATE_ORIGIN,
```
Comment (georgios-uber) [inline modules/core/src/shaderlib/lighting/lighting.js]: What is this syntax?

Comment (Pessimistress) [reply modules/core/src/shaderlib/lighting/lighting.js]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Assigning_to_new_variables_names_and_providing_default_values

**Pull request merged**